src/Device/Driver/FANET/Parser.cpp

```diff
@@ -0,0 +1,304 @@
+  }
+
+  return true;
+}
+
+/*
+ *  Service(type 4 - weather info):
+ *  #FNF src_manufacturer,src_id,broadcast,signature,type,payloadlength,payload
+ */
+bool
+FanetDevice::ParseRxPacket(char *cmd_str, StationsList &stations_list,
+                           double clock) {
+  // Prepare the line
+  cmd_str = RemoveExtraChars(cmd_str);
+  if ( ! IntegrityCheck(cmd_str))
+    return false;
+
+  char *start_ptr = (char *)cmd_str;
+  char *end_ptr;
+
+  if(! start_ptr)
+    return false;
+
+  // Manufacturer(1 byte)
+  uint8_t manufacturer = strtol(start_ptr, &end_ptr, 16);
+  if(start_ptr == end_ptr)
+    return false;
+
+  start_ptr = strchr(start_ptr, ',') + 1;  // comma after src_manufacturer
+  if (start_ptr == nullptr)
+    return false;
+
+  // Id(2 bytes)
+  uint16_t id = strtol(start_ptr, &end_ptr, 16);
+  if(start_ptr == end_ptr)
+    return false;
+
+  /*
+   * Skip broadcast(1 byte) and signature(2 bytes)
+   *  comma after src_id
+   *  comma after broadcast
+   *  comma after signature
+   */
+  for (int i = 0; i < 3; ++i) {
+    start_ptr = strchr(start_ptr, ',') + 1;
+    if (start_ptr == nullptr)
+      return false;
+  }
+
+  // Type(1 byte)
+  uint8_t type = strtol(start_ptr, &end_ptr, 16);
+  if(start_ptr == end_ptr)
+    return false;
+
+  start_ptr = strchr(start_ptr, ',') + 1;  // comma after type
+  if (start_ptr == nullptr)
+    return false;
+
+  // Payload length(1 byte)
+  uint16_t payload_length = strtol(start_ptr, &end_ptr, 16);
+  if(start_ptr == end_ptr)
+    return false;
+
+  /*
+   * Standard limitation of LoRa buffers of 256 bytes and the maximum length
+   * of the FANET MAC header of 11 bytes reduces the permitted payload length
+   * to 245 bytes.
+   */
+  if(payload_length > 245)
+    return false;
+
+  // Payload(payload_length bytes)
+  auto* payload = new uint8_t[payload_length];
+  start_ptr = strchr(start_ptr, ',') + 1;  // comma after payloadlength
+  if (start_ptr == nullptr)
+    return false;
+
+  // Extract payload
+  for(int i = 0; i < payload_length; i++)
+  {
+    char sstr[3] = {start_ptr[i * 2], start_ptr[i * 2 + 1], '\0'};
+    if(strlen(sstr) != 2)
+      return false;
+
+    payload[i] = strtol(sstr,  &end_ptr,  16);
+    if(start_ptr == end_ptr)
+      return false;
+  }
+
+  switch (type)
+  {
+    // We only decode type 4 (service / weather data) for now
+    case FANET_TYPE_SERVICE:
+      FanetStation station;
+      station.address = FanetAddress(manufacturer, id);
+      if(DecodeServicePayload(payload, payload_length, &station))
+        return AddStationToList(stations_list, clock, station);
+      return false;
+    default:
+      return false;
+  }
+}
+
+/*
+ * https://github.com/3s1d/fanet-stm32/raw/master/fanet_module.pdf
+ * https://github.com/3s1d/fanet-stm32/blob/master/Src/fanet/radio/protocol.txt
+ */
+bool
+FanetDevice::ParseNMEA(const char *line, NMEAInfo &info)
+{
+  if(StringIsEqual(line, "#FNF", strlen("#FNF"))) {
```

Comment:
Use `StringAfterPrefix()`, that's much easier and safer than your complicated pointer arithmetic with two `strlen()` calls!